tests/test_incidents.py

```diff
@@ -0,0 +1,183 @@
+        product="hello", settings=None, config=test_config, extrasettings=None
+    )
+    assert "<Incidents product: hello>" == str(inc)
+
+
+def test_incidents_call():
+    """
+    What is the bare minimal set of arguments
+    needed by the callable?
+    """
+    test_config = {}
+    test_config["FLAVOR"] = {}
+    inc = Incidents(product="", settings=None, config=test_config, extrasettings=None)
+    res = inc(incidents=[], token={}, ci_url="", ignore_onetime=False)
+    assert res == []
+
+
+def test_incidents_call_with_flavors():
+    test_config = {}
+    test_config["FLAVOR"] = {"AAA": {"archs": []}}
+    inc = Incidents(product="", settings=None, config=test_config, extrasettings=None)
+    res = inc(incidents=[], token={}, ci_url="", ignore_onetime=False)
+    assert res == []
+
+
+def test_incidents_call_with_incidents():
+    class MyIncident(object):
+        def __init__(self):
+            self.id = None
+            self.staging = False
+            self.livepatch = False
+            self.packages = [None]
+            self.rrid = None
+            self.revisions = {("", ""): None}
+
+        def revisions_with_fallback(self, arch, version):
+            pass
+
+    test_config = {}
+    test_config["FLAVOR"] = {"AAA": {"archs": [""], "issues": {}}}
+    inc = Incidents(
+        product="",
+        settings={"VERSION": "", "DISTRI": None},
+        config=test_config,
+        extrasettings=None,
+    )
+    res = inc(incidents=[MyIncident()], token={}, ci_url="", ignore_onetime=False)
+    assert res == []
+
+
+def test_incidents_call_with_issues():
+    class MyIncident(object):
+        def __init__(self):
+            self.id = None
+            self.staging = False
+            self.livepatch = False
+            self.packages = [None]
+            self.rrid = None
+            self.revisions = {("", ""): None}
+            self.channels = []
+
+        def revisions_with_fallback(self, arch, version):
+            pass
+
+    test_config = {}
+    test_config["FLAVOR"] = {"AAA": {"archs": [""], "issues": {"1234": ":"}}}
+    inc = Incidents(
+        product="",
+        settings={"VERSION": "", "DISTRI": None},
+        config=test_config,
+        extrasettings=None,
+    )
+    res = inc(incidents=[MyIncident()], token={}, ci_url="", ignore_onetime=False)
+    assert res == []
+
+
+@pytest.fixture
+def request_mock(monkeypatch):
+    """
+    Aggregate is using requests to get old jobs
+    from the QEM dashboard.
+    At the moment the mock returned value
+    is harcoded to [{}]
+    """
+
+    class MockResponse:
+        # mock json() method always returns a specific testing dictionary
+        @staticmethod
+        def json():
+            return [{"flavor": None}]
+
+    def mock_get(*args, **kwargs):
+        return MockResponse()
+
+    monkeypatch.setattr("openqabot.types.incidents.requests.get", mock_get)
+
+
+def test_incidents_call_with_channels(request_mock):
+    class MyIncident(object):
```

Comment:
same for MyIncident class - you have a lot of duplicated code here which can be reduced with fixtures

Reply:
I agree for `Incidents` 
I probably disagree for `MyIncident`: each implementation in each test is adding the minimal amount of element needed in the class to reach the point of the Incidents code reached by the test (is it a good way to test it? Maybe not, see my disclaimer https://github.com/openSUSE/qem-bot/pull/143#issuecomment-1738666443 )

I could maybe address this suggestion as an improvement in a next PR

Reply:
sure , it is just "nice to have " so up to you